src/test/java/uk/nhs/ctp/service/ValidationServiceTest.java

```diff
@@ -51,4 +58,164 @@ public void zip_creation() throws IOException {
+  }
+
+  @Test
+  public void zipAudits_withEncounterLackingCaseId_shouldFail() throws IOException {
+    var audits = List.of(AuditSession.builder().build());
+
+    expectedException.expect(UnsupportedOperationException.class);
+    validationService.zipAudits(audits, OperationType.ENCOUNTER);
+  }
+
+  @Test
+  public void zipAudits_shouldOnlyZipGetEntries() throws IOException {
+    var getEntry = AuditEntry.builder()
+        .requestMethod("GET")
+        .dateOfEntry(Instant.now())
+        .requestUrl("http://valid.com/request/url")
+        .responseBody("{}")
+        .build();
+    var audits = List.of(AuditSession.builder()
+        .entry(AuditEntry.builder().requestMethod("POST").build())
+        .entry(AuditEntry.builder().requestMethod("PUT").build())
+        .entry(AuditEntry.builder().requestMethod("DELETE").build())
+        .entry(getEntry)
+        .additionalProperty("caseId", "33")
+        .build());
+
+    var zippedAudits = validationService.zipAudits(audits, OperationType.ENCOUNTER);
+
+    assertThat(unzipEntries(zippedAudits), hasSize(1));
+  }
+
+  @Test
+  public void zipAudits_withEncounterAudits_shouldZipEntryData() throws IOException {
+    var now = Instant.now();
+    var then = now.plus(1, SECONDS);
+    var entry1 = AuditEntry.builder()
+        .requestMethod("GET")
+        .responseBody("{ \"body\": \"validRequestBody1\" }")
+        .requestUrl("http://valid.com/request/url1")
+        .dateOfEntry(now)
+        .build();
+    var entry2 = AuditEntry.builder()
+        .requestMethod("GET")
+        .responseBody("<body>validRequestBody2</body>")
+        .requestUrl("http://valid.com/request/url2")
+        .dateOfEntry(then)
+        .build();
+    var entry3 = AuditEntry.builder()
+        .requestMethod("GET")
+        .responseBody("{ \"body\": \"validRequestBody3\" }")
+        .requestUrl("http://valid.com/request/url3")
+        .dateOfEntry(then)
+        .build();
+
+    var audit1 = AuditSession.builder()
+        .entry(entry1)
+        .entry(entry2)
+        .additionalProperty("caseId", "11")
+        .build();
+    var audit2 = AuditSession.builder()
+        .entry(entry3)
+        .additionalProperty("caseId", "22")
+        .build();
+
+    var audits = List.of(audit1, audit2);
+
+    var zipped = validationService.zipAudits(audits, OperationType.ENCOUNTER);
+
+    assertThat(unzipEntries(zipped), containsInAnyOrder(
+        matchesAuditEntry("encounter11/valid.com/request/url1.json", entry1),
+        matchesAuditEntry("encounter11/valid.com/request/url2.xml", entry2),
+        matchesAuditEntry("encounter22/valid.com/request/url3.json", entry3)
+    ));
+  }
+
+  @Test
+  public void zipAudits_withServiceSearchAudits_shouldZipEntryData() throws IOException {
+    var entry = AuditEntry.builder()
+        .requestMethod("GET")
+        .responseBody("{ \"body\": \"validRequestBody1\" }")
+        .requestUrl("http://valid.com/request/url1")
+        .dateOfEntry(Instant.now())
```

Comment:
So that this works, the clock should be a dependency on the class and mocked.

Reply:
Not sure why we would need a clock here - the `ValidationService` never retrieves the current date-time, it only uses the instant provided in the `AuditEntry` - provided in the tests as the current time. This can be replaced with a fixed/hard-coded `Instant` if you think it helps readability.

Reply:
Yep definitely should be fixed in tests. Sorry, assumed it was getting current date as that's what you were doing here!
As above could it be fixed to an instant truncated to seconds to negate the abstraction from the zip entry process?